lib/screens/joinroom.dart

```diff
@@ -47,9 +48,18 @@ class _JoinRoomState extends State<JoinRoom> {
   ///  - players_list
   ///  - new_game
   /// -------------------------------------------------------------------
-  _onGameDataReceived(Map message) {
-    playersList = (message["data"])["players"];
-    currPlayer = new Player(name: _name.text);
+  _joinRoomListener(Map message) {
+    if (playersList.length == 0) {
+      playersList = (message["data"])["players"];
+      // print(playersList);
+      currPlayer = new Player(name: _name.text);
+      // Assign the ID of the player
+      playersList.forEach((player) {
+        if (player["name"] == _name.text) {
+          currPlayer.id = player["id"].toString();
```

Comment:
I think player id is number in the backend. But there are multiple cases of id as string in frontend.